### PR TITLE
Added recordEvent to Travel pay link.

### PR DESCRIPTION
**Pull request by brianseek**
## Description
Adds recordEvent to travel reimbursement link

## Original issue(s)
department-of-veterans-affairs/va.gov-team#38176

### CODE REVIEW

File: src/applications/check-in/components/TravelPayReimbursementLink.jsx [hunk 1]
```diff
@@ -1,14 +1,25 @@
...
+import React, { useCallback } from 'react';
 import { useTranslation } from 'react-i18next';
+import recordEvent from 'platform/monitoring/record-event';
+
+import { createAnalyticsSlug } from '../utils/analytics';
 import ExternalLink from './ExternalLink';
 
 function TravelPayReimbursementLink() {
   const { t } = useTranslation();
+  const handleClick = useCallback(() => {
```
Comment (acrollet) [inline src/applications/check-in/components/TravelPayReimbursementLink.jsx:10]: What would you think of passing the event ID instead and conditionally recording the event in the ExternalLink component if it's given one? Seems like we're likely to want info on other link clicks and feels a little nicer than passing around a handler to me. (likely only a matter of taste though)

**Review by acrollet — CHANGES_REQUESTED** [review]
Looks like there's a legit unit test failure to deal with :-)

File: src/applications/check-in/components/TravelPayReimbursementLink.jsx (hunk 1 above)
Comment (brianseek) [reply src/applications/check-in/components/TravelPayReimbursementLink.jsx:10]: yeah that makes sense to me!

**Review by acrollet — APPROVED** [review]
I confirmed that the handler is called for the BTSSS link and not others, I like the new approach :)

**Pull request merged**